### ISSUE TITLE
Restore the terms 'Permalink' and 'Slug' in the PostURL user interface

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#61196]
### Description

Splitting this out from https://github.com/WordPress/gutenberg/issues/56381#issuecomment-2061235862

In https://github.com/WordPress/gutenberg/pull/60632 some long-established temrinology regartding the 'post permalink' has been changed. I'm not sure there has been a broader discussion to validate whether these changes are any good for users. I'd liek to start a broader discussion about that.

Current state on trunk, where the new UI uses the term 'Link':

![Screenshot 2024-04-29 at 14 43 43](https://github.com/WordPress/gutenberg/assets/1682452/c3d87bfc-9fb6-4a46-9ea8-170e891b1bc7)

While I'd agree that, if I had to explain tp a complete newbie or to non-tech-savvy users what to edit here, I would use the most simple phrasing e.g. 'Edit the post link', I'd also note that, _for ages_, WordPress has been using its own specific terminology:

## Permalink
The term 'permalink' has been for ages one of the most prominent terminology used in WordPress. For years, it appeared right below the post title, in a very prominent place:

![Screenshot 2024-04-29 at 14 21 32](https://github.com/WordPress/gutenberg/assets/1682452/34f49dce-79c5-44cc-80fb-3c57791b2c01)

Even though it's a somewhat 'technical' term, WordPress users are very familiar with what a 'Permalink' is.

The admin menu uses the same term under Settings > Permalinks at wp-admin/options-permalink.php

## Slug 
The term 'Slug' or 'URL Slug' has been used in WordPress for ages as well. I can count around 30 occurrences of the term 'slug' used within transtlatable strings.

Sometimes this term is used for objects other than a post, such as plugins, widgets, patterns, font collections, etc. Regardless, it's a widely used term and WordPress users are familiar with it.

## Concerns 
https://github.com/WordPress/gutenberg/pull/60632 changed the terms used in the 'edit permalink' user interface after very little discussion. I'm not sure such changes should be done that lightly. I'm not aprioristically opposed to changes but I definitely think _changes that impact long established conventions should be validated by user testing_. In absence of user testing that validates the proposed change performs better, we should just avoid to introduce such changes.

In this specific case, personally I don't see a good reason for this change.

Also, I'd like to point out the new terminology 'Link' is technically incorrect. Both in the settings panel and in the popover, 'Link' is now the label for the _last part of the URL_ However, in this UI, users can edit _the last part of the URL_. they can't edit the 'Link'. In everyday's language, although a little incorrectly, the term 'Link' is used to indicate _the full URL_.

Overall, my question is: _why_ change these terms? In which way this change makes the user experience any better? 

I would say a more wise approach would be to _not_ change long established conventions. Although both terms 'permalink' and 'slug' are a little technical, to me there's no good reason to stop using them. Rather, the Editor UI should a better job to explain new users what they are about.

More screenshots of the admin UI where the Permalink and Slug terms are still in use:

![Screenshot 2024-04-29 at 14 25 26](https://github.com/WordPress/gutenberg/assets/1682452/f19997d7-dfe9-4a3b-bbe4-f15431f08c74)
![Screenshot 2024-04-29 at 14 39 50](https://github.com/WordPress/gutenberg/assets/1682452/c6ed08a9-a843-4fa3-b490-cb70fa27b7f4)
![Screenshot 2024-04-29 at 14 40 04](https://github.com/WordPress/gutenberg/assets/1682452/488011e0-61f9-4106-b5d2-d6f92df4941d)

### Step-by-step reproduction instructions

- Edit a published post (to make sure it has a saved slug).
- In the Post settings panel, observe the slug is now called 'Link'.
- Click the slug to open the popover.
- Observe the popover title is 'Link'.
- Observe the input field has an aria-label attribute 'Link'.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Previously in WordPress 6.5:

The Post panel settings shows, or at least attempts to show, the _full URL_. This is actually what WordPress calls the 'permalink', i.e. the _full_ URL:

In the popover, the term 'permalink' is used incorrectly because that's actually the _slug_:

![Screenshot 2024-04-29 at 15 08 02](https://github.com/WordPress/gutenberg/assets/1682452/46359ec3-2b86-4a75-ba12-49c247724584)

**Comment by afercia** [comment]
For some history: at the beginning of this Gutenberg project, the 'Permalink UI' has been discussed at length. To make an informed decision, I would suggest everyone to do some GitHub archeology and read the discussions on the issues and PRs related to the permalink UI.


The root issue with the permalink UI is that the post URL may be very long. In order to show the full URL, the permalink UI should be placed in a spot that provides _sufficient width_ to show long content. The settings panel is, by its own nature, limited in width.

As a consequence, efforts have been made to attempt to display the permalink in a 'smart way', which resulted in truncating the text. However, that is less than ideal.

In 6.5., this (truncated permalink in the settings panel):

![Screenshot 2024-04-29 at 15 27 25](https://github.com/WordPress/gutenberg/assets/1682452/1cff5ed4-0aff-46fb-b924-7234250fcb81)

Is clearly _less usable_ then the UI in the Classic editor, which does a better job to show long URLs:

![Screenshot 2024-04-29 at 15 28 30](https://github.com/WordPress/gutenberg/assets/1682452/597f1d8f-2725-4fa2-9d90-8b2ab66ea3e4)

Also, the Classic Permalink UI uses some smart handling of long URLs to attempt to display only the relevant parts, as explained in https://github.com/WordPress/gutenberg/issues/61158

Basically, until 6.5 the Gutenberg editor hasn't solved yet this problem and users can't see _the full permalink_ in the UI.

The latest changes on trunk make things worse because the UI now shows only the slug.

There is no way to see the full post permalink.

I would say that, even before any accessibility consideration, this is a serious _usability and design_ issue.

I would like to suggest to use some lateral thinking and try to solve the root issue instead of the symptoms.

Rather than trying to 'squeeze' the content in a limited-width UI, I think it's time to consider to move the permalink UI to a different place. A place in the editor UI that provides way more horizontal space. Clearl, not a side panel.

**Comment by afercia** [comment]
Additionally, also the toggle button that opens the PostURL popover is labeled incorrectly.

Imagine you're editing a post with title 'Hello World'.
As such, the permalink of your post is `https://myawesomesite.org/hello-world/`.

The toggle button that opens the popover will be labeled with `aria-label="Change link: hello-world"`, which is clearly incorrect because _'hello-world' is not a link_.

**Comment by annezazu** [comment]
@afercia just coming to this as I agree it's harder to edit the link of a post in the new experience and have added it to the new tracking issue for writing experience.

**Comment by afercia** [comment]
Thank you @annezazu 👋🏽 

The UI is slightly changed on latest trunk, still the terminology in use could be improved.

I'd also like to note that the 'Learn more' link leads to a [documentation page](https://wordpress.org/documentation/article/page-post-settings-sidebar/#permalink) that is oud of date. It still shows the old UI and it should be updated.

Screenshot of the latest UI on trunk:

![Screenshot 2024-07-09 at 10 09 12](https://github.com/WordPress/gutenberg/assets/1682452/fb60a7ff-2d6a-4a6e-b318-8f8895f21254)

Screenshot of the documentation page that needs to be updated:

![Screenshot 2024-07-09 at 10 14 29](https://github.com/WordPress/gutenberg/assets/1682452/dd503a50-5c66-4c17-8bdc-ff3d3cbdcee4)

**Comment by joedolson** [comment]
I really don't see why this issue is still lingering; the label is just wrong as is, and needs to be updated. I feel like it's driven by a desire not to use words that require technical insight to fully understand, such as 'slug'; but that's a disservice to users, as we're instead using a word that is *not what they're editing*. 

It is not difficult to learn what a slug is; the text that currently explains that a user is actually only editing the last part of the link would be perfectly accurate as a description of 'slug', instead.

**Comment by afercia** [comment]
To my understanding, the only case where this control shows a _link_ is when setting the WordPress home page to use a static page (a 'front page'). However, in this case I'd argue this row should only _display_ the link and not open a popover with a link that can't be edited. Screenshot:

![Screenshot 2024-07-17 at 12 46 30](https://github.com/user-attachments/assets/c556037a-858d-4c7d-a0f3-a470b1ccac68)

**Comment by afercia** [comment]
Just noticed: passing the whole link [as the help prop of the input field](https://github.com/WordPress/gutenberg/blob/1eff14a6d9d04a3a85c61889cb6422f005853adb/packages/editor/src/components/post-url/index.js#L143-L159) is very arguable and highlights there is no understanding about what a description for an input filed is supposed to be.

An input field description, associated with aria-describedby, is meant to provide a meaningful explanation of what the input field is about, or instructions and hints. It's not a place where to use a value. I'm guessing this has been done only for visual purposes, to place the link below the input field. Semantically, it's just plain wrong.

![Screenshot 2024-07-17 at 13 30 14](https://github.com/user-attachments/assets/597a49fb-2c57-44b2-afb8-252f16e21406)

**Comment by afercia** [comment]
One more thing to fix: when the compoent shows the non-editable _front page link_, the aria-label still says it can be 'changed' even if it's not editable: `aria-label="Change link: front-page"`.

**Comment by t-hamano** [comment]
The "Link" label can be even more confusing depending on the permalink settings.

For example, select "Custom Structure" as the permalink structure and enter `/%post_id%/`. Then create a post with the title "Hello World".

The button text in the panel will show `/hello-world`, and in the popover it will show `http://localhost:8888/{post_id}/`. Even though the button text is a slug and not part of the permalink, because it is labeled "Link" users will believe this text is part of the permalink, and will be further confused because the button text and the true permalink URL in the popover are completely different.

![link](https://github.com/user-attachments/assets/952fa288-0643-4f90-87d0-10b208edcf14)

**Comment by t-hamano** [comment]
I would like to propose the following UI. All permalink structures have the following in common:

- Change the label from "Link" to "Permalink"
- Toggle button text displays the permalink in a single line, hiding overflow text
- Change the title of the popover from "Link" to "Permalink"
- Give the input field in the popover a visible label: "Slug"
- Remove the custom grey color of the permalink in the popover
- Move the copy button from the input field to next to the permalink (Related to [#63700](https://github.com/WordPress/gutenberg/issues/63700))
- Move the help text about the slug below the input field
- Permalinks are always editable, just like in the classic theme.

### Custom Structure: Post name

#### trunk

![image](https://github.com/user-attachments/assets/4bbbee74-b459-4193-88cd-2298f342e2e1)

#### Proposed UI

![image](https://github.com/user-attachments/assets/cb120d58-bd3f-4bed-a17c-3e90614cba5b)

### Custom Structure: Custom (/%post_id%/)

#### trunk 

![image](https://github.com/user-attachments/assets/0450efec-227e-49d8-8352-fe49384ec49d)

#### Proposed UI

Note that there is no "Customize the last part of the URL." text because when the custom structure is a post id, the slug is not part of the URL.

![image](https://github.com/user-attachments/assets/be141726-f59e-419b-b680-210c8ec27e8a)

**Comment by afercia** [comment]
@t-hamano good point. And it's not limited to the `{post_id}` example. There are other bits of the permalink that may be non-editable. For example, a custom structure like:

`/%category%/%post_id%/%author%/`

In this case, the Classic editor only shows the Permalink with no UI to edit (because there's nothing that can be edited):

![Screenshot 2024-07-22 at 10 02 02Intead, w](https://github.com/user-attachments/assets/ad7cc958-a77e-4c0d-ba6f-b5e6e84dd667)

Instead, when the permalink contains an editable part, the editing UI does show up:

![Screenshot 2024-07-22 at 10 01 04](https://github.com/user-attachments/assets/dafa5235-e90b-4800-881b-3a5a671c3ab0)

I'd think you're right and the UI logic should be reversed: the Permalink should be the value that is always shown, whether it is editable or not. The slug may be part of the Permalink and should be shown only when it is editable.

The only problematic point I see is that the Inspector panel isn't the right place to show the full permalink because it may be very long. As a user, I want to see the full permalink to check it's a good one that make sense for my post. Which brings us back to old discussions for example in https://github.com/WordPress/gutenberg/issues/577 and in  https://github.com/WordPress/gutenberg/issues/5494. Still, I'm not convinced the Edit permalink UI should live in a narrow panel.

**Comment by t-hamano** [comment]
@afercia Thanks for the feedback.

> In this case, the Classic editor only shows the Permalink with no UI to edit (because there's nothing that can be edited):

This is a bit confusing, but the slug is editable in the classic editor even if it's not part of the permalink.

If you open the Screen Options and check "Slug", an input field will appear below the content editor:

![image](https://github.com/user-attachments/assets/1a5ae4ca-6f17-41a0-92a2-c2b854fe1e57)

So in my opinion, the slug should _always_ be editable in the block editor, whether it's part of the permalink or not.

But maybe this should be discussed in a separate issue.

**Comment by t-hamano** [comment]
This issue may be related to #56381.

**Comment by afercia** [comment]
> So in my opinion, the slug should always be editable in the block editor, whether it's part of the permalink or not.

Yes, this is something to consider.

Overall, I think the current UI doesn't provide users with the best experience. Weirdest case is when the permalink structure does _not_ contain the `/%postname%/`. In that case:
- The panel shows a value e.g. `my-post-title` labeled 'Link'. However, it's not a link and can't be edited.
- Clicking `my-post-title` opens the popover that shows the non-editable permalink, and it doesn't contain ``my-post-title`` of course.

![Screenshot 2024-07-26 at 11 48 19](https://github.com/user-attachments/assets/37607dc1-f17a-4b5a-bced-c029776e3547)

**Comment by richtabor** [comment]
> Change the label from "Link" to "Permalink"

Is it necessary to call it two different things: permalinks and slugs? I don't thing the panel should be different from the input (visual label or not).

**Comment by richtabor** [comment]
Seems Slug is a better fit than Permalink.

**Comment by t-hamano** [comment]
> Is it necessary to call it two different things: permalinks and slugs?

The terms "Permalink" and "Slug" need to be used clearly.

An example of the definition of the terms is quoted from [this webpage](https://www.dreamhost.com/glossary/seo/slug/):

> - Slug: Slugs are the last part of a path and represent a unique piece of content – they help identify the content so that visitors can more easily remember and find it.
> - Permalink: Permalinks (or permanent links) are the URLs for posts, pages and other pieces of content on your website, including their paths and slugs. Permalinks never change, meaning you can always link back to them no matter how many times you edit your page or post.

**Comment by richtabor** [comment]
Right, so in this case the label should be Slug, right? It’s confusing calling the setting in the Inspector Permalink, but you edit the Slug.

**Comment by joedolson** [comment]
It makes sense to me; what you're being shown (the URL), is the post permalink; but the portion of it you can edit is the slug. So the popover can reasonably have a heading for 'Permalink', since it's showing you what the permalink is for the post, but the input field should be slug.

**Comment by t-hamano** [comment]
> in this case the label should be Slug, right? It’s confusing calling the setting in the Inspector Permalink, but you edit the Slug.

This may be admittedly confusing, but I think the label (not the popover title) should be "Permalink". Because if the permalink doesn't contain a slug, the slug doesn't mean much:

If we change the label to "Slug", the UI would look like this:

![image](https://github.com/user-attachments/assets/313f4518-fb30-4467-975c-99c7b25692be)

Additionally, I find it odd that the permalink is displayed in a popover that opens when I click a button labeled "Slug".

I'm still leaning towards the UI change suggested in [this comment](https://github.com/WordPress/gutenberg/issues/61196#issuecomment-2241610484).

**Comment by joedolson** [comment]
I'm comfortable with the UI change suggested there, @t-hamano.

**Comment by richtabor** [comment]
I'm not confident just yet. What's the formal proposal to change it to?

**Comment by t-hamano** [comment]
I've moved forward with what was suggested in [this comment](https://github.com/WordPress/gutenberg/issues/61196#issuecomment-2241610484) and pushed it to the [enhance-post-url-ui](https://github.com/WordPress/gutenberg/tree/enhance-post-url-ui) branch.

The changes are as follows:

- Change the label from "Link" to "Permalink"
- Toggle button text displays the permalink in a single line, hiding overflow text
- Change the title of the popover from "Link" to "Permalink"
- Give the input field in the popover a visible label: "Slug"
- Remove the slash prefix from the input field
- Remove the custom grey color of the permalink in the popover
- Move the copy button from the input field to next to the permalink (Related to [#63700](https://github.com/WordPress/gutenberg/issues/63700))
- Move the help text about the slug below the input field

These changes take into account the proposed designs below, but could be improved upon:

- https://github.com/WordPress/gutenberg/pull/63706#issuecomment-2239417982
- https://github.com/WordPress/gutenberg/issues/56381#issuecomment-1843255480
- https://github.com/WordPress/gutenberg/issues/56381#issuecomment-1847685455

### Permalink structure: Post name

![image](https://github.com/user-attachments/assets/727b5fe3-97fe-4360-abba-f3a748b5434b)

### Permalink structure: Day and name

![image](https://github.com/user-attachments/assets/93ae8763-00ac-4ffa-92c2-f19227534772)

### Permalink structure: Plain

![image](https://github.com/user-attachments/assets/7e6850c0-ddab-45ba-a7af-1ca836b690f5)

**Comment by richtabor** [comment]
It does seem a disadvantage to have `http://localhost:888...` rather than `/hello-world` visible in the Inspector though.

**Comment by joedolson** [comment]
I think there are some other comments elsewhere about what to show in link editing, and those might be worth reviewing. I think we tended to come down on needing to see the unique portions of a URL more than the non-unique portions; e.g., showing the slug in the Inspector would make sense to me, as well. 

The permalink label is informing us that this will edit the permalink; since we can't possibly see the entire URL in that space, it makes more sense to me to see the slug. 

This change might have been driven by Andrea's comments in https://github.com/WordPress/gutenberg/issues/61196#issuecomment-2084577651; but I think the intention there is to change the aria-label, not the visible label. E.g. instead of "Change link: /hello-world", "Change slug: /hello-world". There are still outstanding issues about using the value as the button text; but [there is a separate issue for that](https://github.com/WordPress/gutenberg/issues/63992).

**Comment by t-hamano** [comment]
So, should we keep the labels and button text as they are and focus on improving the popover in this issue?

In other words, the UI will look like this:

![image](https://github.com/user-attachments/assets/33fb5dd0-bf13-4252-9161-3755888b5478)

I think we can discuss the labels and button text in #63992.

**Comment by joedolson** [comment]
I assuming you're referring to the text 'Link' in the post summary controls as the Label in this context; I still think that should change to Permalink. That'll provide textual consistency between the panel and the popover.
